builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -421,9 +421,16 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Default:  "scsi",
 							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 								value := v.(string)
-								if value != "scsi" && value != "ide" {
```

Comment:
I think we need to do something about this ValidateFunc. We need to refactor it to something like declaring a []string of values and then checking that the value is in the slice. The cyclomatic complexity of if && && && etc doesn't really read the best :)